app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -795,7 +795,7 @@ def prov_set_show_vars
       @options = @miq_request.options
       @options[:memory], @options[:mem_typ] = reconfigure_calculations(@options[:vm_memory][0]) if @options[:vm_memory]
       @force_no_grid_xml   = true
-      @view, @pages = get_view(Vm, :view_suffix => "VmReconfigureRequest", :where_clause => ["vms.id IN (?)", @miq_request.options[:src_ids]]) # Get the records (into a view) and the paginator
+      @view, @pages = get_view(Vm, :view_suffix => "VmReconfigureRequest", :selected_ids => @miq_request.options[:src_ids]) # Get the records (into a view) and the paginator
```

Comment:
is this working?
```
get_view(Vm.where(:id =>  @miq_request.options[:src_ids] ) , :view_suffix => "VmReconfigureRequest")
```

Reply:
Nope, that won't work. `get_view` requires a model (`ApplicationRecord`) as first parameter, passing  a query method (`ActiveRecord::QueryMethods::WhereChain`) is not possible.

The `get_view` is defined [here](https://github.com/ManageIQ/manageiq-ui-classic/blob/7fd18a099ce2789ee00a1a6a9a75b02b31cfa43e/app/controllers/application_controller.rb#L1515). The first action over the first parameter is a [conversion to string](https://github.com/ManageIQ/manageiq-ui-classic/blob/7fd18a099ce2789ee00a1a6a9a75b02b31cfa43e/app/controllers/application_controller.rb#L1521), which is later used to calculate strings, etc...

Reply:
thanks  @tumido 👍

Reply:
@lpichler we should probably enhance the get_view method to eat the relation or arel

Reply:
yes it would be nice and then it can go to RBAC call directly. ✂️ parameters!

Reply:
@lpichler, @Ladas: actually no.

the "model" is later called "db" and it ends in `MiqReport.load_from_view_options(db, current_user, options, db_view_yaml_cache)`

There it is used to calculate FILE NAME of a report that is later loaded.

So forget about arel here.

Reply:
Guys, please, leave any larger rearch to a later time. Right now we need the scopes to get rid of the where clauses.

W/o it we are in deep trouble on the UI side right now.

Reply:
....and the `db` / `model` is actually going to come over HTTP so it needs to be a string. This is to be a hight level HTTP API for the reports to feed the GTL component. So we don't want to talk Ruby/arel to it. We want to talk JSON.

Reply:
hum, yeah, well this was the future talk, we need to clean it up first